core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/aggregator/Aggregator.java

```diff
@@ -0,0 +1,295 @@
+/**
+ * 
+ */
+package edu.uci.ics.texera.dataflow.aggregator;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.uci.ics.texera.api.constants.ErrorMessages;
+import edu.uci.ics.texera.api.dataflow.IOperator;
+import edu.uci.ics.texera.api.exception.TexeraException;
+import edu.uci.ics.texera.api.field.DoubleField;
+import edu.uci.ics.texera.api.field.IntegerField;
+import edu.uci.ics.texera.api.field.IField;
+import edu.uci.ics.texera.api.schema.Attribute;
+import edu.uci.ics.texera.api.schema.AttributeType;
+import edu.uci.ics.texera.api.schema.Schema;
+import edu.uci.ics.texera.api.schema.Schema.Builder;
+import edu.uci.ics.texera.api.tuple.Tuple;
+import edu.uci.ics.texera.dataflow.common.AbstractSingleInputOperator;
+import edu.uci.ics.texera.dataflow.comparablematcher.ComparablePredicate;
+
+/**
+ * @author avinash
+ * 
+ * Aggregator operator is used to perform aggregation (like sum, count, min, max, average) operations on a column.
+ *
+ */
+public class Aggregator extends AbstractSingleInputOperator
+{
+    private final AggregatorPredicate predicate;
+    
+    private Schema inputSchema;
+    private Schema outputSchema;
+
+    public Aggregator(AggregatorPredicate predicate)
+    {
+        this.predicate = predicate;
+    }
+    
+    @Override
+    protected void setUp() throws TexeraException
+    {
+        inputSchema = inputOperator.getOutputSchema();
+        List<AggregationAttributeAndResult> aggregationItems = predicate.getAttributeAggregateResultList();
+        for(int i=0; i<aggregationItems.size(); i++)
+        {
+            Schema.checkAttributeExists(inputSchema, aggregationItems.get(i).getAttributeName());
+            Schema.checkAttributeNotExists(inputSchema, aggregationItems.get(i).getResultAttributeName());
+        }
+
+        for(int i=0; i<aggregationItems.size(); i++)
+        {
+            if(aggregationItems.get(i).getAggregatorType() != AggregationType.COUNT)
+            {
+                if(Schema.isAttributeType(inputSchema, aggregationItems.get(i).getAttributeName(), AttributeType._ID_TYPE)
+                   || Schema.isAttributeType(inputSchema, aggregationItems.get(i).getAttributeName(), AttributeType.DATE)
```

Comment:
I think we should support `min, max` for `STRING` and `TEXT`, `min, max, average` for `DATE`